### PR TITLE
internal/engine/local: controller for local serve

### PR DESCRIPTION
**Pull request by dbentley**
This is a draft.

Things that are missing that should be here before merging:
* integrate with DI/wire
* upper/reducer logic

For a next PR:
* actually run processes

### CODE REVIEW

File: internal/engine/local/controller.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+				SequenceNum:  sequenceNum,
+				Status:       status,
+			})
+		}
+	}()
+
+	proc.doneCh = c.execer.Start(ctx, spec.ServeCmd, logger.Get(ctx).Writer(logger.InfoLvl), statusCh)
+}
+
+// lastProcess represents the last process
```
Comment (jazzdan) [inline internal/engine/local/controller.go]: Can we expand on this comment? Why do we need a sentinel struct to know when we've reached the end of the processes?

Comment (dbentley) [reply internal/engine/local/controller.go]: Oh, it's not a sentinel struct. It's where we store info about the most recent exec, if any. Ideas for better names?

Comment (jazzdan) [reply internal/engine/local/controller.go]: Ohh I see. `mostRecentProcess`? Or the extra verbose, non-Go-like `mostRecentlyExecedProcess`?

Comment (dbentley) [reply internal/engine/local/controller.go]: Went for currentProcess, but happy to change.

**Pull request merged**